projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php

```diff
@@ -127,13 +129,114 @@ protected function user_has_access( $access_level, $is_blog_subscriber, $is_paid
 		}
 
 		if ( $access_level === self::POST_ACCESS_LEVEL_PAID_SUBSCRIBERS ) {
-			return $is_paid_subscriber;
+			return $is_paid_subscriber && ! $this->maybe_gate_access_for_user_if_tier( $post_id, $user_abbreviated_subscriptions );
 		}
 
 		// This should not be a use case
 		return false;
 	}
 
+	/**
+	 * Check post access for tiers.
+	 *
+	 * @param int   $post_id Current post id.
+	 * @param array $user_abbreviated_subscriptions User subscription abbreviated.
+	 *
+	 * @return bool
+	 */
+	private function maybe_gate_access_for_user_if_tier( $post_id, $user_abbreviated_subscriptions ) {
+		$tier_id = intval(
+			get_post_meta( $post_id, META_NAME_FOR_POST_TIER_ID_SETTINGS, true )
+		);
+
+		if ( ! $tier_id ) {
+			return false;
+		}
+
+		$plan_ids = \Jetpack_Memberships::get_all_newsletter_plan_ids();
+
+		if ( ! in_array( $tier_id, $plan_ids, true ) ) {
+			// If the tier is not in the plans, we bail
+			return false;
+		}
+
+		// We now need the tier price and currency, and the same for the annual price (if available)
+		$tier_meta         = get_post_meta( $tier_id );
+		$tier_price        = $tier_meta['jetpack_memberships_price'][0];
+		$tier_currency     = $tier_meta['jetpack_memberships_currency'][0];
+		$tier_product_id   = $tier_meta['jetpack_memberships_product_id'][0];
+		$annual_tier_price = $tier_price * 12;
+
+		// At this point we know the post is
+		$linked_post_tier = get_posts(
+			array(
+				'posts_per_page' => 1,
+				'fields'         => 'ids',
+				'post_type'      => \Jetpack_Memberships::$post_type_plan,
+				'meta_key'       => 'jetpack_memberships_tier',
+				'meta_value'     => $tier_product_id,
+			)
+		);
+
+		$annual_tier_id = false;
+		if ( count( $linked_post_tier ) !== 0 ) {
+			$annual_tier_id = (int) reset(
+				$linked_post_tier
+			);
+
+			$annual_tier_meta  = get_post_meta( $annual_tier_id );
+			$annual_tier_price = isset( $annual_tier_meta['jetpack_memberships_price'][0] ) ? $annual_tier_meta['jetpack_memberships_price'][0] : $annual_tier_price;
+		}
+
+		foreach ( $user_abbreviated_subscriptions as $subscription_plan_id => $details ) {
+			$end = is_int( $details->end_date ) ? $details->end_date : strtotime( $details->end_date );
+			if ( $end < time() ) {
+				// subscription not active anymore
+				continue;
+			}
+
+			$subscription_post_id = get_posts(
+				array(
+					'posts_per_page' => 1,
+					'fields'         => 'ids',
+					'post_type'      => \Jetpack_Memberships::$post_type_plan,
+					'meta_key'       => 'jetpack_memberships_product_id',
+					'meta_value'     => $subscription_plan_id,
+				)
+			);
+
+			if ( empty( $subscription_post_id ) === 0 ) {
```

Comment:
We can probably simplify that check?

```suggestion
			if ( empty( $subscription_post_id ) ) {
```

Reply:
Fixed